server/auth/gitlab/gitlab.go

```diff
@@ -223,3 +241,13 @@ func (c *Config) Username(token *oauth2.Token) string {
 	}
 	return u.Username
 }
+
+// sliceOfGroups returns a list of groups that were passed to this function as a comma separated string
+func sliceOfGroups(groups string) []string {
+	spaceTrimmed := strings.TrimSpace(groups)
```

Comment:
This will only trim space at the start and end - anything with spaces around the commas like "group1, group2" won't get caught: https://go.dev/play/p/uMZRq0wodOi

How about this?
```go
func sliceOfGroups(groups string) []string {
	sliced := strings.Split(groups, ",")
	for i := range sliced {
		sliced[i] = strings.TrimSpace(sliced[i])
	}
	return sliced
}
```